### PR TITLE
asc-15 : add typing event in chat

### PR DESCRIPTION
**Pull request by avivt10**

### CODE REVIEW

**Comment by avivt10** [comment]
@Shaybush  Please give me a review

File: client_react/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "react": "^18.2.0",
     "react-bootstrap": "^2.9.1",
     "react-dom": "^18.2.0",
+    "react-loader-spinner": "^5.4.5",
```
Comment (Shaybush) [inline client_react/package.json:15]: why react loader relevant to this task

File: client_react/src/components/Socket/socket.jsx [hunk 2]
```diff
@@ -25,6 +25,19 @@ export default function Socket() {
     };
   });
 
+  useEffect(() => {
+    socket.on("typing-from-server", (id) => {
+      // checking with the user ID
```
Comment (Shaybush) [inline client_react/src/components/Socket/socket.jsx:30]: can replace comment with - "checking if received user typing is current user"

File: client_react/src/components/Socket/socket.jsx [hunk 3]
```diff
@@ -47,6 +60,10 @@ export default function Socket() {
     socket.emit("clientObjEvent", item);
   };
 
+  const handleInput = () => {
+    socket.emit("typing",userData.id);
+  };
+
```
Comment (Shaybush) [inline client_react/src/components/Socket/socket.jsx:66]: can remove and write it in the JSX

File: client_react/src/components/Socket/socket.jsx [hunk 4]
```diff
@@ -67,9 +84,13 @@ export default function Socket() {
           })}
         </div>
         <form onSubmit={onSub} className="chat-form bg-white p-2 d-flex align-items-center justify-content-center">
-          <input ref={inputRef} className="form-control me-1" id="id_input" placeholder='Type here...' />
+          <input onChange={handleInput} ref={inputRef} className="form-control me-1" id="id_input" placeholder='Type here...' />
```
Comment (Shaybush) [inline client_react/src/components/Socket/socket.jsx:87]: `onChange={() => socket.emit("typing", userData.id)}`

File: client_react/src/components/Socket/socket.jsx [hunk 5]
```diff
@@ -67,9 +84,13 @@ export default function Socket() {
...
           <button className="btn btn-dark">Send</button>
         </form>
+        {/* typing */}
+        {
+         typing ? <h6>typing...</h6> : null
+        }
```
Comment (Shaybush) [inline client_react/src/components/Socket/socket.jsx:93]: better replace it above input. try to avoid UI bugs

File: server_node/app.js [hunk 6]
```diff
@@ -4,6 +4,8 @@ const express = require("express");
 const path = require("path");
 // aloud us to create service
 const http = require("http");
+//increasing memory
```
Comment (Shaybush) [inline server_node/app.js:7]: add space after slushes (//)

**Review by Shaybush — CHANGES_REQUESTED** [review]
please change the files I mentioned

File: client_react/package.json (hunk 1 above)
Comment (avivt10) [reply client_react/package.json:15]: The library is used in the loader component.

File: client_react/src/components/Socket/socket.jsx [hunk 7]
```diff
@@ -83,14 +81,17 @@ export default function Socket() {
             );
           })}
         </div>
+          <div className="d-flex m-2 mb-0">
```
Comment (Shaybush) [inline client_react/src/components/Socket/socket.jsx:84]: redundant div

File: client_react/src/index.css [hunk 8]
```diff
@@ -123,3 +123,8 @@ p {
 /* borders - end */
 
 /* colors - end */
+
+/* typing style */
+.px-5{
```
Comment (Shaybush) [inline client_react/src/index.css:128]: why override this bootstrap class ?

**Review by Shaybush — CHANGES_REQUESTED** [review]
please change

**Review by Shaybush — APPROVED** [review]

**Pull request merged**